corehq/apps/userreports/operators.py

```diff
@@ -47,6 +48,15 @@ def in_(input, reference):
     return input in reference
 
 
+def regex_search(input, reference):
+    try:
+        regex = re.compile(reference)
+    except re.error:
+        raise BadSpecError(f"Invalid regular expression: {reference}")
```

Comment:
it would be great if this could happen at parse time (it would only be possible if the reference expression was a constant)

Reply:
I agree, that would definitely be ideal. I felt sort of lucky that raising an error here put it out in a useful way at all (in generic inbound responses and in the expression debugger).